### PR TITLE
Get all articles #8

### PR DESCRIPTION
**Pull request by ojcorrigan**
Created new endpoint GET /api/articles
gets articles and sorts them in ascending order,
counts the number of comments for each article, 
tested and working, 
also testing for 404 not found as was the only error I could think of for this.

### CODE REVIEW

**Comment by Liam310** [comment]
Hey Owen! Going to leave a few comments on the code here, and then one final one with any general points/just to let you know I've finished my commenting 🙂

File: __tests__/app.test.js [hunk 1]
```diff
@@ -52,68 +52,106 @@ describe('getTopics', () => {
...
+  
+  describe('getArticles', () => {
+    test('200 GET /api/articles responds with an array of article objects', () => {
+      return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then((result) => {
+        expect(result.body).toBeInstanceOf(Array)
+        expect(result.body.forEach((article) => {
+          expect(article).toMatchObject({
```
Comment (Liam310) [inline __tests__/app.test.js]: Remember to check that the array that comes back has a length that is more than zero - otherwise an empty array might pass your tests, since what's inside a `forEach` never gets run for an empty array!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -52,68 +52,106 @@ describe('getTopics', () => {
...
+      })
+      })
+      test('/api/article responds 404 not found', () => {
+        return request(app)
+          .get('/api/article')
+          .expect(404)
+          .then((result) => {
+            expect(result.body.msg).toBe('invalid path');
+          });
+      });
```
Comment (Liam310) [inline __tests__/app.test.js]: For now, this does seem like the only thing that can wrong on this endpoint, but I would say you've already handled it previously! The 404 you did for `/api/topic` is a generic "this an incorrect path/typo/spelling mistake" scenario where we only need to consider it once. Just the one test for this error will do it - no need to consider lots of typos!

File: controllers/article-controllers.js [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+    next(err);
+  });
+};
+
+exports.getArticles = (req,res,next) => {
+  
+  selectArticles().then((result) => {
+    res.status(200).send(result)
+  })
+}
```
Comment (Liam310) [inline controllers/article-controllers.js]: I think this is a bit of a trend in your controllers (well, I suppose I should say in your tests since they decide what your code must do!) of just sending back the result from the model. The result being that the `res.body` you get back in the tests is itself an array, when really what we want is to wrap the array in an object, so our response is not:
```
[
  { ... },
  { ... }
]
```

but is instead:

```
{
  articles: [
    { ... },
    { ... }
  ]
}
```

I would take the opportunity, once you reach the end of a ticket, to do a bit of a refactor in both the tests and controllers to account for this little change (and it is little!)

**Comment by Liam310** [comment]
Okay - that's all my comments! Looks like a lot of good stuff, just a few minor things to pick up on at an opportune moment 🙂
